riscv/src/runtime.rs

```diff
@@ -276,39 +276,76 @@ impl Runtime {
         self.syscalls.contains_key(&s)
     }
 
-    pub fn with_poseidon(mut self) -> Self {
+    fn with_poseidon(mut self, continuations: bool) -> Self {
+        let init_call = if continuations {
```

Comment:
what are these mstore for?

Reply:
in the bootloader case, the first time an address is accessed needs to be a bootloader write. Since we call `poseidon_gl 0, 0;` as the init call for the dummy block, we need to make sure all the addresses used for input and output in the poseidon machine have been accessed by the bootloader first